### PR TITLE
Supported weighted hexbin plots

### PR DESCRIPTION
**Pull request by larkery**
This amendment allows stat_binhex to support weighted data by getting the assignment of input points to hexagons out of the hexbin function, and then gluing that to an input weight vector to find a weighted sum for each hexagon.

### CODE REVIEW

File: R/stat-binhex.r [hunk 1]
```diff
@@ -88,12 +89,30 @@ hexBin <- function(x, y, binwidth) {
...
+  
+  if (!is.null(weight)) {
+      ## we need to produce weights for each cell rather than counts. hb@cID contains the
+      ## ID of the cell for each input point and hb@cell contains the vector of cell IDs
+      ## which relate to values in hb@count.
+      
+      ## consequently we can put cID next to weight, and then sum them grouped by cID to get
+      ## total weight for each bin
+      sumWeights <- data.frame(cellID = hb@cID, weight=weight)
+      sumWeights <- aggregate(. ~ cellID, data=sumWeights, FUN=sum)
```
Comment (hadley) [inline R/stat-binhex.r]: I think you could do this without an intermediate data frame by using `tapply()`

File: R/stat-binhex.r [hunk 2]
```diff
@@ -88,12 +89,30 @@ hexBin <- function(x, y, binwidth) {
...
+      
+      ## finally we want to join the weights back onto the "compressed" list of cells
+      ## hb@cell contains the indices that have actually been used, in the order they occur.
+      count <- sumWeights[match(hb@cell, sumWeights$cellID), c('weight')]
```
Comment (hadley) [inline R/stat-binhex.r]: I think this would be easier to read as `weight[match(hb@cell, sumWeights$cellID)]`

**Comment by hadley** [comment]
Looks good.  Could you please:

1. [ ] Respond to the changes described above
1. [ ] Add a bullet to NEWS
1. [ ] Add a unit test

**Comment by hadley** [comment]
If you're interested in working on this please ping me, and I'll reopen (I'm closing to make it easier for me to manage the release)

**Comment by larkery** [comment]
I will do these things; I am not entirely sure how to do a unit test for this, but I will dig around and see what the standard is - cheers.

**Pull request closed without merging**